nhs/data/geography.py

```diff
@@ -0,0 +1,295 @@
+        df.to_pandas(),
+        geometry=gpd.points_from_xy(df["x"], df["y"]),
+    )
+    projected_crs = CRS.from_string(crs)
+    return gdf.to_crs(projected_crs)
+
+
+def process_chunk(
+    map_data: gpd.GeoDataFrame,
+    chunk: gpd.GeoDataFrame,
+    area_column: str = "SA2_NAME21",
+    area_code_column: str = "SA1_CODE21",
+) -> pl.DataFrame:
+    """
+    Process a chunk of point data, spatially joining it with area polygon data.
+
+    Parameters
+    ----------
+    map_data : gpd.GeoDataFrame
+        A GeoPandas DataFrame containing spatial polygon data that represents areas (e.g., districts or regions).
+        This data includes geometry (polygons) and area-specific attributes such as 'SA2_NAME21' and 'SA1_CODE21',
+        which represent area names and codes.
+
+    chunk : gpd.GeoDataFrame
+        A GeoPandas GeoDataFrame containing point data (e.g., latitude and longitude) that needs to be joined with
+        the corresponding areas in `map_data`. This should contain a 'geometry' column representing points.
+
+    Returns
+    -------
+    pl.DataFrame
+        A Polars DataFrame with the point data from the `chunk`, joined with the appropriate area data from
+        `map_data`. The output includes the original coordinates ('x', 'y') and the corresponding area names
+        ('SA2_NAME21') and area codes ('SA1_CODE21'), renamed as 'area' and 'area_code', respectively.
+
+    Notes
+    -----
+    - The function first performs a spatial join using the 'within' predicate to match points with polygons
+      in the `map_data`. This returns the areas where each point is located.
+    - If any points cannot be matched with an area (i.e., points that do not fall within any polygon), a
+      nearest neighbor spatial join is performed to assign the closest area.
+    - The final result includes only the 'x', 'y' coordinates and the corresponding area information, and it
+      is returned as a Polars DataFrame.
+    - The function converts the intermediate GeoDataFrame to a dictionary before constructing a Polars DataFrame
+      to ensure compatibility with the Polars framework.
+    - The name is a SA2 variable as SA1s don't have names. They do of course have codes, though, which we use.
+    """  # noqa: E501
+    pnts_with_area = gpd.sjoin(chunk, map_data, how="left", predicate="within")
+    missing_points = pnts_with_area[pnts_with_area["SA2_NAME21"].isnull()]
+
+    if not missing_points.empty:
+        nearest_join = gpd.sjoin_nearest(
+            missing_points[["geometry"]], map_data, how="left"
+        )
+        pnts_with_area.loc[missing_points.index, "SA2_NAME21"] = nearest_join[
+            "SA2_NAME21"
+        ]
+        pnts_with_area.loc[missing_points.index, "SA1_CODE21"] = nearest_join[
+            "SA1_CODE21"
+        ]
+
+    result = pnts_with_area[["x", "y", "SA2_NAME21", "SA1_CODE21"]]
+    result = result.rename(columns={"SA2_NAME21": "area", "SA1_CODE21": "area_code"})
```

Comment:
Looks like you are assuming that the columns will have certain name like `"SA2_NAME21"`? Consider having them as parameters with those names as default value and add in docstring that the dataframes need columns with these names. If we work with dataframes with standardised names then the function will fail